src/modules/universal-discovery/components/create/choose.language.component.js

```diff
@@ -26,6 +26,17 @@ export default class ChooseLanguageComponent extends Component {
             selectedLanguageCode = props.allowedLanguages[0];
         }
 
+        if (selectedLanguageCode === null || !props.languages.mappings[selectedLanguageCode].enabled) {
+            const languages = Object.values(props.languages.mappings);
+            const filteredLanguages = languages.filter((language) => {
+                const isAllowedLanguage = props.allowedLanguages.length ? props.allowedLanguages.includes(language.languageCode) : true;
+
+                return isAllowedLanguage && language.enabled;
+            });
+
+            selectedLanguageCode = filteredLanguages.length ? filteredLanguages[0].languageCode : '';
```

Comment:
I think if `selectedLanguageCode` will be `''` the UDW will crash.

Reply:
it is not possible, because the only one where language is verified via the state for content creation is in the `renderOptions` function, but it will not be performed as well because no option will be available or enabled and form can be submit

Reply:
After this component is mouted it updates the selectedLanguage in parent component. The parent component checks permissions and may do other stuff. It would be safer to never set `selectedLanguage` to null, better would be `{}`.